enhancing/enhancer.spec.js

```diff
@@ -1,2 +1,223 @@
+      expect(succeed(1)).toBeNull();
+      expect(succeed("abc")).toBeNull();
+    });
+
+    it("should not increase enhancement if already at 20", () => {
+      const testObject = {
+        name: "Thor",
+        durability: 99,
+        enhancement: 20
+      };
+      expect(succeed(testObject)).toEqual(testObject);
+    });
+  });
+
+  // FAIL MODULE TEST
+  describe('the fail module', () => {
+    it('should decrease the durability by 5 if the existing enhance value is less than 15', () => {
+      const testItem = {
+        name: "Black Wizard",
+        durability: 75,
+        enhancement: 14
+      }
+
+      const returnItem = {
+        name: "Black Wizard",
+        durability: 70,
+        enhancement: 14
+      }
+
+      expect(fail(testItem)).toEqual(returnItem)
+    })
+
+    it('should decrease the enhancement by 1 if the existing enhance value is greater than 16', () => {
+      const testItem = {
+        name: "Black Panther",
+        durability: 75,
+        enhancement: 17
+      }
+
+      const returnItem = {
+        name: "Black Panther",
+        durability: 65,
+        enhancement: 16
+      }
+
+      expect(fail(testItem)).toEqual(returnItem)
+    })
+
+    it('should decrease the durability by 10 if the existing enhancement value is greater than 14', () => {
+      const testItem = {
+        name: "Black Panther",
+        durability: 75,
+        enhancement: 15
+      }
+
+      const returnItem = {
+        name: "Black Panther",
+        durability: 65,
+        enhancement: 15
+      }
+
+      expect(fail(testItem)).toEqual(returnItem)
+    })
+
+    it("it should return null or undefined when not passed a valid object", () => {
+      const improperObject = {
+        name: "Thor",
+        speed: 50,
+        enhancement: 15
+      };
+
+      //expect(repair([1])).toBe(expected);
+      expect(fail(1)).toBeNull();
+      expect(fail("a")).toBeNull();
+      expect(fail(improperObject)).toBeUndefined();
+      expect(fail([])).toBeUndefined();
+    });
+
+    it("it should return null when called with no arguments", () => {
+      // assert
+      expect(fail()).toBeNull();
+    });
+  })
+});
+
+describe('the get module', () => {
+  it('should leave the name alone if enhancement value is 0', () => {
+    const item = {
+      name: "Iron Man",
+      durability: 98,
+      enhancement: 0
+    }
+
+    expect(get(item)).toEqual(item);
+  })
+
+  it('should prefix the name with the enhanced value if enhancement value is > 0', () => {
+    const item = {
+      name: "Iron Man",
+      durability: 98,
+      enhancement: 7
+    }
+
+    const expected = {
+      name: "[+7] Iron Man",
+      durability: 98,
+      enhancement: 7
+    }
+
+    expect(get(item)).toEqual(expected);
+  })
+
+  it("it should return null when called with no arguments", () => {
+    // assert
+    expect(get()).toBeNull();
+  });
+
+  it("it should return null or undefined when not passed a valid object", () => {
```

Comment:
This might be clearer if worded:
`"it should return null or undefined when passed an invalid object"`

Reply:
The same goes for others worded in a negative way (over a positive way)!